RileyLinkKitTests/PumpOpsSynchronousBuildFromFramesTests.swift

```diff
@@ -0,0 +1,250 @@
+//  PumpOpsSynchronousBuildFromFramesTests.swift
+//  RileyLink
+//
+//  Created by Jaim Zuber on 3/8/17.
+//  Copyright © 2017 Pete Schwamb. All rights reserved.
+//
+
+import XCTest
+
+@testable import RileyLinkKit
+import MinimedKit
+import RileyLinkBLEKit
+
+class PumpOpsSynchronousBuildFromFramesTests: XCTestCase {
+    
+    var sut: PumpOpsSynchronous!
+    var pumpState: PumpState!
+    var pumpID: String!
+    var pumpRegion: PumpRegion!
+    var rileyLinkCmdSession: RileyLinkCmdSession!
+    var pumpModel: PumpModel!
+    var pumpOpsCommunicationStub: PumpOpsCommunicationStub!
+    
+    override func setUp() {
+        super.setUp()
+        
+        pumpID = "350535"
+        pumpRegion = .worldWide
+        pumpModel = PumpModel.Model523
+        
+        rileyLinkCmdSession = RileyLinkCmdSession()
+        pumpOpsCommunicationStub = PumpOpsCommunicationStub(session: rileyLinkCmdSession)
+        
+        loadSUT()
+    }
+    
+    func loadSUT() {
+        pumpState = PumpState(pumpID: pumpID, pumpRegion: pumpRegion)
+        pumpState.pumpModel = pumpModel
+        pumpState.awakeUntil = Date(timeIntervalSinceNow: 100) // pump is awake
+        
+        sut = PumpOpsSynchronous(pumpState: pumpState, session: rileyLinkCmdSession)
+        sut.communication = pumpOpsCommunicationStub
+    }
+    
+    func testErrorIsntThrown() {
+        pumpOpsCommunicationStub.responses = buildResponsesDictionary()
+        
+        assertNoThrow(try _ = sut.getHistoryEvents(since: Date()))
+    }
+    
+    func testUnexpectedResponseThrowsError() {
+        var responseDictionary = buildResponsesDictionary()
+        var pumpAckArray = responseDictionary[.getHistoryPage]!
+        pumpAckArray.insert(sut.makePumpMessage(to: .buttonPress), at: 0)
+        responseDictionary[.getHistoryPage]! = pumpAckArray
+        
+        pumpOpsCommunicationStub.responses = responseDictionary
+        
+        // Didn't receive a .pumpAck short reponse so throw an error
+        assertThrows(try _ = sut.getHistoryEvents(since: Date()))
+    }
+    
+    func testUnexpectedPumpAckResponseThrowsError() {
+        var responseDictionary = buildResponsesDictionary()
+        var pumpAckArray = responseDictionary[.getHistoryPage]!
+        pumpAckArray.insert(sut.makePumpMessage(to: .buttonPress), at: 1)
+        responseDictionary[.getHistoryPage]! = pumpAckArray
+        
+        pumpOpsCommunicationStub.responses = responseDictionary
+        
+        // Didn't receive a .getHistoryPage as 2nd response so throw an error
+        assertThrows(try _ = sut.getHistoryEvents(since: Date()))
+    }
+    
+    func testAllEventsReturned() {
+        pumpOpsCommunicationStub.responses = buildResponsesDictionary()
+        
+        let date = Date(timeIntervalSince1970: 0)
+        do {
+            let eventsTuple = try sut.getHistoryEvents(since: date)
+            let historyEvent = eventsTuple.0
+            
+            // Ends because of out of order
```

Comment:
The name of the test is testAllEventsReturned, but this comment seems to indicate it's returning early when encountering an out of order event?